Polyhedron/demo/Polyhedron/MainWindow.cpp

```diff
@@ -2369,6 +2366,79 @@ void MainWindow::propagate_action()
     }
   }
 }
+
+void MainWindow::on_actionSa_ve_Scene_as_Script_triggered()
+{
+  QString filename =
+      QFileDialog::getSaveFileName(this,
+                                   "Save the Scene as a Script File",
+                                   last_saved_dir,
+                                   "*.js");
+  std::ofstream os(filename.toUtf8());
+  if(!os)
+    return;
+  std::vector<QString> names;
+  std::vector<QString> loaders;
+  std::vector<QColor> colors;
+  std::vector<int> rendering_modes;
+  for(int i = 0; i < scene->numberOfEntries(); ++i)
+  {
+    Scene_item* item = scene->item(i);
+    QString loader = item->property("loader_name").toString();
+    QString source = item->property("source filename").toString();
+    if(loader.isEmpty())
+      continue;
```

Comment:
Maybe the list of items that cannot be saved should be pushed into a list, so that a popup can explain which items are not saved.